### ISSUE TITLE
[Updated Request] please update "Flexget" to latest and take out of beta

### ISSUE THREAD

**Issue by alirz1** [SynoCommunity/spksrc#2551]
Please have "Flexget" package updated in the synocommunity repo. It can also be taken out of beta as it works perfectly without any problems.

**Comment by alirz1** [comment]
Does anyone monitor these requests?

**Comment by KaraokeStu** [comment]
Hi yes, we do monitor these requests, however we do this for free and have other things that also require our time. One of the issues we have is that this particular project updates a release every few days, but does not seem to have any form of "self update" facility.

You may be better (if you want to be running the latest and greatest) to follow the opkg instructions here:
https://flexget.com/InstallWizard/SynologyNAS

**Comment by cytec** [comment]
Also if you'd installed python SPK you could generate a virtualenv and `pip install flexget` there if you'd like to. AFAIK flexget doesn't require any packages that require cross compilation

**Comment by alirz1** [comment]
@cytec Thanks for the info. But this is the issue.

Uptil now i've been the using the Optware method of installing the latest version of flexget. That has been workng fine. However as optware project itself seems to be coming to an end, i would like to move to a method that would allow me to keep using newer version of flexget.

Currently i have python and python3 installed on my synology from the community repo. However i dont find the "virtualenv" path at /usr/local/python/bin/virtualenv or anywhere else on the system.

Also "pip" doesnt seem to get installed as part of the two above mentioned python packages already installed on my box. At this point im just trying to figure out the most straight forward way of getting latest flexget installed.

**Comment by cytec** [comment]
should look like this if you have the synocommunity SPK installed:

    daenerys> ls /usr/local/python3/bin/
    2to3               bzfgrep            delgroup           infocmp            pip3               reset              tset               xz
    2to3-3.4           bzgrep             deluser            infotocap          pip3.4             shuf               unlink             xzcat
    addgroup           bzip2              dumpsexp           lzcat              pydoc3             speexdec           unlzma             xzdec
    adduser            bzip2recover       easy_install       lzma               pydoc3.4           speexenc           unxz
    bunzip2            bzless             easy_install-3.4   lzmadec            python3            sqlite3            virtualenv
    busybox            bzmore             fatattr            lzmainfo           python3.4          start-stop-daemon  virtualenv-3.4
    bzcat              c_rehash           gpg-error          mpicalc            python3.4m         tabs               wheel
    bzcmp              captoinfo          hmac256            nice               pyvenv             tic                xmlcatalog
    bzdiff             clear              idle3              openssl            pyvenv-3.4         toe                xmllint
    bzegrep            curl               idle3.4            pip                renice             tput               xsltproc

however if you are looking for a more `optware|ipkg` way... you might try `entware-ng|opkg` https://github.com/Entware-ng/Entware-ng/wiki/Install-on-Synology-NAS

**Comment by alirz1** [comment]
@cytec Hi, Let me ask you a dumb question.. When you say "if you have the synocommunity SPK installed".What exactly do you mean?

**Comment by Dr-Bean** [comment]
People, let's stop with discussing various flavors of optware and/or manual installations here. This is a bugtracker for SynoCommunity packages, let's keep it focused on that subject. There are better places to discuss other installation methods, for example Synology forums or the software's own support forums.

@alirz1 Unfortunately, the active SynoCommunity maintainers don't seem to have enough bandwidth to keep the package up-to-date at this time. If you're interested in an up-to-date package, you'll either have to wait until contributors do have time, or perhaps you'd like to see if you can update the package yourself. If you're not sure how to, but would like to give it a go, feel free to ask questions.

**Comment by alirz1** [comment]
Thanks. This issue can be closed.

**Comment by Dr-Bean** [comment]
Well...we don't mind keeping the issue open. It serves as a reminder to update the package, and shows contributors what they can contribute to :)
So, if you don't mind, I'll reopen.

**Comment by ymartin59** [comment]
I give it a try...

**Comment by ymartin59** [comment]
Package almost ready... I am building a XPEnology DSM 5.2 for testing. Pull request should come soon.

**Comment by alirz1** [comment]
looking forward to this. I had building the apl. While the flexget apk gets built it was only 32KB in size, which doesnt sense.
Would you provide the apk for download. 
And if you dont mind would you share you build setup?

**Comment by cytec** [comment]
@alirz1 well i guess @ymartin59 is creating a spk with/for spksrc and opens a PR afterwards, so it should be available on synocommunity after it was merged and also in spksrc for everyone to compile it on their own if they'd like to

**Comment by ymartin59** [comment]
@Dr-Bean @Diaoul I am annoyed with current package versioning because 20150605-5 is considered as "upper" version than "2.9.11-6" I have planned to used as new package version. As package was beta, does it sound acceptable to ask users to uninstall "beta" package to be able to install new package with now proper versioning ?

**Comment by ymartin59** [comment]
@Dr-Bean @Diaoul An other "newbie" question: instead of copy/pasting "requirements.txt" from GitHub repository, I would like to add a make target to do it automatically before "wheel" processing. Is there any pre_wheel "hook" I may define (and how) to do so ? Thank you in advance

**Comment by Dr-Bean** [comment]
@ymartin59 Wrt versioning: Not sure I care all that much. It's either not being able to present the correct version in Package Center, or instruct people to reinstall. The problem with the reinstall is that people who have the current package installed will not get an update presented via Package Center...

Wrt requirements.txt:
The problem with requirements.txt is that the are no 'rules' on what it should contain. Some projects add all the requirements (including dependencies of dependencies), and others add just top-level requirements.
The last category is especially problematic to us, because for packages we _must_ include _all_ dependencies, doubly so because we don't have access to a compiler on the device.

Since you cannot be sure upfront if all the requirements are included by upstream, you'll have to make sure of it. Blindly copying won't be any good :)
An additional challenge is that dependency processing of Python packages isn't great, so you end up with workarounds or half-baked solutions whatever you do. For example, there are some tools (`pip-tools` comes to mind, and there are others) that can process an existing requirements.txt and add and update all dependencies, but all of those tools have limitations. We can't set up a generic solution at this time...believe me, I've looked into it, it simply does not exist, and people way smarter than me are working on it.

So, currently, the only correct way to be sure that the package's requirements.txt is complete is to install the project and recreate a new requirements.txt based off of that. See the WIKI article on wheels, I've explained a couple of options.
Personally, I find it easiest to use a Debian-based VM (separate from dev env to ensure no cross-contamination can occur with dependencies), install the project manually into a virtualenv and test the software, then create a requirements.txt based off of that.
